File: group-34-interledger-API/Models/FileMetaData.cs
using group_34_interledger_API.Enums;

namespace group_34_interledger_API.Models
{
    public class FileMetaData : BaseDataModel
    {
        public string FileName { get; set; }
        public double FileSize { get; set; }
        public FileTypes FileTypes { get; set; }
        public List<FileAccessed> FileAccessedList { get; set; } = new List<FileAccessed>();
    }
}

File: group-34-interledger-NodeJsAPI/Models/FileMetaData.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, CreateDateColumn } from 'typeorm';
import { FileAccessed } from './FileAccessed';

export enum FileTypes {
    Photo = 0,
    Video = 1,
    Document = 2
}

@Entity()
export class FileMetaData {
    @PrimaryGeneratedColumn()
    Id!: number;

    @Column()
    FileName!: string;

    @Column()
    FileUrl!: string;

    @Column()
    FileSize!: number;

    @Column({
        type: "enum",
        enum: FileTypes,
        default: FileTypes.Document
    })
    FileTypes!: FileTypes;

    @OneToMany(() => FileAccessed, fileAccessed => fileAccessed.fileMetaData)
    fileAccesses!: FileAccessed[];

    @CreateDateColumn({ type: 'timestamp' })
    createdAt!: Date;
}
